index.js

```diff
@@ -2405,7 +2416,7 @@
     Date: Date_,
     Error: Error_,
     FiniteNumber: FiniteNumber,
-    Function: def('Function', {}, [Array_(Type), Type], Function_),
+    Function: def('Function', {}, [Array_(VariadicType), Type], Function_),
```

Comment:
I think this should remain unchanged. It doesn't make sense to define `Array -> Array` or `Either -> Either`.